### PR TITLE
adds auth0 authentication for angular

### PR DESCRIPTION
**Pull request by NNKamel**

### CODE REVIEW

File: src/app/auth.service.ts [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    }, () => { }, () => {
+    });
+  }
+
+  // call the users api to get the current user
+  // call this when
+  private tryRetrieveUser() {
+    // call to user service to call user api should go here.
+    // this code should be used once services and apis are ready
+
```
Comment (clarson217) [inline src/app/auth.service.ts]: The commented out code is preparation for future merges.

File: src/app/auth.service.ts [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    //   console.log("error getting user data" + err.message);
+    //   this.isAdmin$.next(false);
+    // });
+  }
+
+  // send a request to check if user is an admin
+  // if user isn't then set isadmin to false
+  private isUserAdmin() {
+    // this code should be used once services and apis are ready
+
```
Comment (clarson217) [inline src/app/auth.service.ts]: The commented out code is preparation for future merges.

**Review by clarson217 — APPROVED** [review]

**Review by TrisTenuto — APPROVED** [review]

File: src/app/interceptor.service.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+})
+
+// injects the token automatically in the request and passes it along
+// only injects the request if user is currenly loggedin
+export class InterceptorService implements HttpInterceptor {
+  loggedIn: boolean;
+
+  constructor(private auth: AuthService) {
+    auth.userProfile$.subscribe((reply) => {
+      console.log("in interceptor: " + reply);
```
Comment (markmoorerev) [inline src/app/interceptor.service.ts]: any console.log()'s need to be taken out for the production version and/or changed to a logged comment

**Review by markmoorerev — CHANGES_REQUESTED** [review]
change console.log()'s to logging statements.

**Pull request closed without merging**